azure/defaults.go

```diff
@@ -383,6 +385,54 @@ func (p userAgentPolicy) Do(req *policy.Request) (*http.Response, error) {
 	return req.Next()
 }
 
+// NewDefaultCredential creates a new credential that uses the following methods, in order of precedence:
+// 1. Workload Identity
```

Comment:
should this be AAD Pod identity since we're not using workload identity anywhere in the main branch yet ?

Reply:
If we go via the chained identity mechanism of sdk, I think we should keep the order as it is specified in the sdk as of now i.e 
env > workload identity > managed identity > cli creds

Reply:
> So I think we can have wrapper over SDK that picks and choose what to use as what is specified in the AzureClusterIdentity.

I like this approach, because the SDK doesn't make it easy to reimplement the default cred chained logic properly. Several items we would want to reuse are not exported. What I did here is a "clean room" reimplementation because it's probably legally problematic to actually copy and modify the SDK sources here.